ci/azure-install-rust.yml

```diff
@@ -0,0 +1,70 @@
+steps:
+  - bash: |
+      set -ex
+      toolchain=$TOOLCHAIN
+      if [ "$toolchain" = "" ]; then
+        toolchain=nightly
+      fi
+      if command -v rustup; then
+        rustup update $toolchain
+        rustup default $toolchain
+      else
+        curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain $toolchain
+        echo "##vso[task.prependpath]$HOME/.cargo/bin"
+      fi
+    displayName: Install rust (unix)
+    condition: ne( variables['Agent.OS'], 'Windows_NT' )
+  - script: |
+      @echo on
+      if not defined TOOLCHAIN set TOOLCHAIN=nightly
+      rustup update %TOOLCHAIN%-%TARGET%
+      rustup default %TOOLCHAIN%-%TARGET%
+    displayName: Install rust (windows)
+    condition: eq( variables['Agent.OS'], 'Windows_NT' )
+  - script: |
+      set -ex
+      if [ -n "${TARGET}" ]; then
+        rustup target add $TARGET
+      fi
+    condition: ne( variables['Agent.OS'], 'Windows_NT' )
+    displayName: Install target (unix)
+  - script: |
+      @echo on
+      if defined TARGET rustup target add %TARGET%
+    condition: eq( variables['Agent.OS'], 'Windows_NT' )
+    displayName: Install target (windows)
+  - bash: |
+      set -ex
+      gcc -print-search-dirs
+      find "C:\ProgramData\Chocolatey" -name "crt2*"
+      find "C:\ProgramData\Chocolatey" -name "dllcrt2*"
+      find "C:\ProgramData\Chocolatey" -name "libmsvcrt*"
+    condition: eq( variables['Agent.OS'], 'Windows_NT' )	
+    displayName: Find GCC libraries (windows)
+  - script: |	
+      @echo on	
+      if defined MSYS_BITS for %%I in (crt2.o dllcrt2.o libmsvcrt.a) do xcopy /Y "C:\ProgramData\Chocolatey\lib\mingw\tools\install\mingw64\x86_64-w64-mingw32\lib\%%I" "%USERPROFILE%\.rustup\toolchains\%TOOLCHAIN%-%TARGET%\lib\rustlib\%TARGET%\lib" 
```

Comment:
Failed with path: `C:\Users\VssAdministrator\.rustup\toolchains\-x86_64-pc-windows-gnu\lib\rustlib\x86_64-pc-windows-gnu\lib`
`%TOOLCHAIN%` is undefined :man_shrugging: